### PR TITLE
Allows sticky grenades to be applied to self from your hand

### PR DESCRIPTION
**Pull request by Scarlet-Oduko**
## About The Pull Request

allows stickies to be applied to the user, and only the user, from in hand
## Why It's Good For The Game

Quality of life and makes the stickies such as cloaker much more usable, and trailblazer to be used offensively by a marine who may be wearing fire armor and such. 
## Changelog
:cl:
qol: Allows grenades to be self stickied, instead of having to be applies by others
balance: Self-applied stickies allowing one to use them directly
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Hiln jetpack suicide bomber meta yo.

**Comment by Scarlet-Oduko** [comment]
would be quite comical, if that is a concern can just remove it on explosive stickies specifically, generally just wanting this for trails and the cloaker.

**Comment by Lumipharon** [comment]
> would be quite comical, if that is a concern can just remove it on explosive stickies specifically, generally just wanting this for trails and the cloaker.

That would likely be best

**Comment by Scarlet-Oduko** [comment]
> > would be quite comical, if that is a concern can just remove it on explosive stickies specifically, generally just wanting this for trails and the cloaker.
> 
> That would likely be best

Kay Kay, I’ll sort that out tomorrow when I can.

**Comment by chizzy376** [comment]
> > would be quite comical, if that is a concern can just remove it on explosive stickies specifically, generally just wanting this for trails and the cloaker.
> 
> That would likely be best

The people fear the taliban meta

**Review by Lumipharon — CHANGES_REQUESTED** [review]
why copy paste code? just make it at the sticky level and have a var/flag that allows the behavior

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 1]
```diff
@@ -137,6 +157,16 @@
 	var/turf/T = get_turf(src)
 	T.ignite(25, 25)
 
+/obj/item/explosive/grenade/sticky/trailblazer/afterattack(atom/target, mob/user)
+	. = ..()
+	if(target != user)
+		return
+	if(self_sticky == FALSE)
+		return
```
Comment (Lumipharon) [inline code/game/objects/items/explosives/grenades/marines.dm:165]: why the copy paste checks?
make the parent return true if it self sticks, then just make an early return on !.

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 2]
```diff
@@ -99,6 +101,23 @@
...
+	user.drop_held_item()
+	activate()
+	var/image/stuck_overlay = image(icon, user, initial(icon_state) + "_stuck")
+	stuck_overlay.pixel_x = rand(-5, 5)
+	stuck_overlay.pixel_y = rand(-7, 7)
+	user.add_overlay(stuck_overlay)
+	forceMove(user)
+	saved_overlay = stuck_overlay
+	stuck_to = user
+	RegisterSignal(stuck_to, COMSIG_QDELETING, PROC_REF(clean_refs))
```
Comment (Lumipharon) [inline code/game/objects/items/explosives/grenades/marines.dm:119]: You're still copy pasting a lot.
This whole thing should just be some 'stuck_to' proc or something, then just add to that as required for the child types.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/explosives/grenades/marines.dm (hunk 1 above)
Comment (Scarlet-Oduko) [reply code/game/objects/items/explosives/grenades/marines.dm]: > why the copy paste checks? make the parent return true if it self sticks, then just make an early return on !.

pushed some more stuff, this is the best I can do with my knowledge, I'd need walked through how to make it any cleaner, messed around with it for awhile and just ended up making it worse for awhile

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 3]
```diff
@@ -83,13 +83,11 @@
 	var/atom/stuck_to
 	///Current image overlay applied to stuck_to, used to remove the overlay after detonation.
 	var/image/saved_overlay
+	///if this specific grenade should be allowed to self sticky
```
Comment (Lumipharon) [inline code/game/objects/items/explosives/grenades/marines.dm:86]:
```suggestion
	///If a mob can stick this to themselves
```

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 4]
```diff
@@ -83,13 +83,11 @@
...
+	var/self_sticky = FALSE
 
-/obj/item/explosive/grenade/sticky/throw_impact(atom/hit_atom, speed)
-	. = ..()
-	if(!.)
-		return
-	if(!active || stuck_to || isturf(hit_atom))
-		return
+///handles sticky overlay and attaching the grenade itself to the target
+/obj/item/explosive/grenade/sticky/proc/stuck_to(atom/hit_atom)
```
Comment (Lumipharon) [inline code/game/objects/items/explosives/grenades/marines.dm:90]: new procs need to be placed after old procs.

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Looking good

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**